src/ui/components/Main/Keys/general/PrivateKeyModal/PrivateKeyModal.tsx

```diff
@@ -0,0 +1,90 @@
+import { ModalGroup } from '../general/ModalGroup/ModalGroup.tsx';
+import cn from './PrivateKeyModal.module.css';
+import { TextAreaGroup } from '../../../../general/TextAreaGroup/TextAreaGroup.tsx';
+import { useForm, useWatch } from 'react-hook-form';
+import { yupResolver } from '@hookform/resolvers/yup';
+import { createSchema } from './schema.ts';
+import { useStoreEffect, useStoreState } from '../../../../../../react-vault';
+import { Button } from '../../../general/Button/Button.tsx';
+import addIcon from '../../../../../assets/addIcon.svg';
+import { KeyPair } from 'near-api-js';
+import { MessageGroup } from '../general/MessageGroup/MessageGroup.tsx';
+
+export const PrivateKeyModal = ({ isOpen, close, setStep }) => {
+  const addKey = useStoreEffect((store: any) => store.keys.addKey);
+  const records = useStoreState((store: any) => store.keys.records);
+  const schema = createSchema(records);
+
+
+  const form = useForm({
+    resolver: yupResolver(schema),
+    defaultValues: {
+      publicKey: null,
+      privateKey: null,
+      seedPhrase: null,
+    },
+  });
+
+  const {
+    control,
+    register,
+    handleSubmit,
+    setValue,
+    resetField,
+    formState: { errors, dirtyFields },
+  }: any = form;
+
+  const publicKey = useWatch({ control, name: 'publicKey' });
+
+  const prevStep = () => {
+    setStep('selectImport');
+  };
+
+  const onSubmit = (data: any) => {
+    const pk = KeyPair.fromString(data.privateKey).getPublicKey().toString();
```

Comment:
Why do you split logic between the effect and the local func? Can we keep all logic in effect?